crates/astria-telemetry/src/macros.rs

```diff
@@ -0,0 +1,58 @@
+// re-export so that they need not be imported by downstream users.
+// hidden because they shouldn't be imported.
+#[doc(hidden)]
+pub use const_format::{
+    concatcp as __concatcp,
+    map_ascii_case as __map_ascii_case,
+    Case as __Case,
+};
+
+/// Declare a `const` string slice, using the declaring crate's name as a
+/// prefix and the variable name as a suffix.
+///
+/// This macro essentially performs this declaration:
+/// ```text
+/// METRIC_NAME := ${CARGO_CRATE_NAME}_metric_name;
+/// ```
+///
+/// The purpose of this macro is to avoid accidental typos, ensuring that the
+/// metric name matches the const variable name.
+///
+/// # Examples
+/// ```
+/// use astria_telemetry::declare_metric_const;
+/// declare_metric_const!(pub const EXAMPLE_COUNTER);
+/// // Note that this example has `astria_telemetry` a a prefix because
+/// // this doctest is part of this crate.
+/// // In your case, use your crate's `CARGO_CRATE_NAME` as prefix.
+/// assert_eq!(EXAMPLE_COUNTER, "astria_telemetry_example_counter");
+/// ```
+#[macro_export]
+macro_rules! declare_metric_const {
```

Comment:
Maybe `metric_name` might be a more succinct way of describing what the macro should be used for?  Or even `declare_metric_name` if you think the term "declare" is needed?

Reply:
Renamed to `metric_name` in [e64ca43](https://github.com/astriaorg/astria/pull/1129/commits/e64ca43d377e20cad7621e12f11821b979e5dd14)